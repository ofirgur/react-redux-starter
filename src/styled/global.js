import { createGlobalStyle } from 'styled-components';
import styledNormalize from 'styled-normalize';

export default createGlobalStyle`
    html {
        box-sizing: border-box;
    }

    *,
    *::before,
    *::after {
        box-sizing: inherit;
    }
    
    ${styledNormalize};
    
    body {
        background: ${props => props.theme.background.body};
        font-family: ${props => props.theme.typography.fontFamity};
        color: ${props => props.theme.color.main};
        word-break: break-word; 
        font-size: ${props => props.theme.typography.mobileFontSize};

        @media only screen and (min-width: ${props => props.theme.breakpoint.sm}) {
            font-size: ${props => props.theme.typography.fontSize}; 
        }

        i {
            &:hover {
                cursor: pointer;
                transform: scale(1.5);
                opacity: .8; 
            }
        }

        a.simple-link {
            color: ${props => props.theme.color.link};
            text-decoration: none;

            &:hover { 
                color: ${props => props.theme.color.hoveredLink};
                text-decoration: underline;
            }
        }
    }
`;