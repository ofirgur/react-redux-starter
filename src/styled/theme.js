export default {
    typography: {
        fontFamily: 'Roboto", "Helvetica", "Arial", sans-serif',
        fontSize: '16px',
        mobileFontSize: '10px'
    },
    breakpoint: {
        xs: '0px',
        sm: '600px',
        md: '960px', 
        lg: '1280px',
        xl: '1920px'
    },
    zIndex: {

    },
    direction: {
        ltr: 'ltr',
        rtl: 'rtl'
    },
    background: {
        body: '#263238',
        main: '#607d8b',
        mainHover: '#78909C'
    },
    color: {
        main: '#fff',
        purple: '#651fff',
        red: '#ff7043',
        link: '#00695C',
        hoveredLink: '#004D40'
    },
    padding: {
        main: '16px',
        mobile: '6px'
    },
    margin: {
        main: '16px',
    },
    shadow: {
        main: '0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)'
    }
};