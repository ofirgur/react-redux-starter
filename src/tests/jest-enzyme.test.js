import React from 'react';
import styled from 'styled-components';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const sum = (x, y) => x + y;
const Button = styled.button`
    background-color: blue;
`;

describe('Jest & Enzyme tests', () => {

    it('should sum correctly', () => {
        expect(sum(3, 7)).toBe(10);
    });

    it('should render correctlly', () => {
        const wrapper = shallow(<Button />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should styled correctly', () => {
        const tree = renderer.create(<Button />).toJSON()
        expect(tree).toMatchSnapshot()
    });
});