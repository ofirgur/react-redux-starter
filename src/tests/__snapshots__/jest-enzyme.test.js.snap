// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest & Enzyme tests should render correctlly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    background-color: blue;
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Jest & Enzyme tests should styled correctly 1`] = `
.c0 {
  background-color: blue;
}

<button
  className="c0"
/>
`;
